apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java

```diff
@@ -88,7 +104,15 @@ public int getStackTraceLimit() {
         return stackTraceLimit.get();
     }
 
-    public long getSpanFramesMinDurationMs() {
-        return spanFramesMinDurationMs.getValue().getMillis();
+    public long getSpanStackTraceMinDurationMs() {
+        if (!spanFramesMinDurationMs.isDefault()) {
+            long spanFramesMinDurationMsValue = spanFramesMinDurationMs.getValue().getMillis();
+            if (spanFramesMinDurationMsValue == 0) {
+                return -1;
+            } else if (spanFramesMinDurationMsValue == -1) {
```

Comment:
Didn't the old config option collect traces for any negative value, not just -1? Does this need to be
`            } else if (spanFramesMinDurationMsValue < 0) {
`

Reply:
fixed